services/lumberjack/src/service.js

```diff
@@ -0,0 +1,255 @@
+  }
+
+  /** Start the service. */
+  async start() {
+    logger.debug('Starting service.');
+
+    /** Database configuration */
+    this._influx = new InfluxDB({
+      host: this._influxHost,
+      port: this._influxPort,
+      database: 'lumberjack',
+      schema: [
+        {
+          measurement: 'ping',
+          fields: {
+            apiPing: FieldType.INTEGER,
+            devicePing: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port',
+            'name'
+          ]
+        },
+        {
+          measurement: 'upload-rate',
+          fields: {
+            t1: FieldType.INTEGER,
+            t5: FieldType.INTEGER,
+            f1: FieldType.INTEGER,
+            f5: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port'
+          ]
+        },
+        {
+          measurement: 'telemetry',
+          fields: {
+            gstatus: FieldType.INTEGER,
+            pstatus: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port'
+          ]
+        }
+      ]
+    });
+    //Create database
+    this._influx.createDatabase('lumberjack');
+
+    this._server = await this._createApi(this._influx);
+    this._startTasks();
+    logger.debug('Service started');
+  }
+
+  /** Stop the service. */
+  async stop() {
+    logger.debug('Stopping service.');
+
+    await Promise.all([
+      this._server.closeAsync(),
+      this._pingTask.stop(),
+      this._uploadRateTask.stop(),
+      this._telemetryTask.stop()
+    ]);
+
+    logger.debug('Service stopped.');
+  }
+
+  // Create the koa api and return the http server.
+  async _createApi(influx) {
+    const app = new Koa();
+
+    app.use(koaLogger());
+
+    app.context.influx = influx;
+
+    // Set up the router middleware.
+    app.use(router.routes());
+    app.use(router.allowedMethods());
+
+    // Start and wait until the server is up and then return it.
+    return await new Promise((resolve, reject) => {
+      const server = app.listen(this._port, (err) => {
+        if (err) reject(err);
+        else resolve(server);
+      });
+
+      // Add a promisified close method to the server.
+      server.closeAsync = () => new Promise((resolve) => {
+        server.close(() => resolve());
+      });
+    });
+  }
+
+  _startTasks() {
+    this._pingTask =
+      createTimeoutTask(this._pingTask.bind(this), this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._uploadRateTask =
+      createTimeoutTask(this._uploadRate.bind(this), this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._telemetryTask =
+      createTimeoutTask(this._telemetryOverview.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _pingTask() {
+    // Get ping data
+    let ping =
```

Comment:
need to `try`-`catch` this, when pong is down you should essentially record a blank in influx